VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/api/VTCareResource.java

```diff
@@ -20,4 +23,30 @@ public Provider createProvider(Provider provider) throws Exception {
             throw new Exception("An exception occurred while persisting the provider details.");
         }
     }
+
+    @POST
+    @Path("login")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response login(String json) throws Exception {
+        JsonObject o = new Gson().fromJson(json, JsonObject.class);
+        try {
+            String email = o.get("email").getAsString();
+            String password = o.get("password").getAsString();
+            boolean isProviderLogin = o.get("isProviderLogin").getAsBoolean();
+
+            LoginService loginService = new LoginService();
+            int id = loginService.login(email, password,
```

Comment:
'id' gives no context to the variable. It should make it clear which type of object is it identifying. Rename it appropriately.

Reply:
I believe in the context of logging in, refers to the id of the user. It is generic to a patient and provider, and adding them in the name would just not make it worthwhile.

Reply:
You know it, I agree. if at this stage it's not clear if it would be a provider or patient then use userId. 
Id can be as it is in a for example when it is used in class there it is unambiguous, or in a db table for that matter. In this context we must use a more readable name for variables.

Reply:
Think it from the perspective of understandability and maintainability. Those are important concerns in any software. For someone else reading your code, it should be absolutely clear what a particular variable stores. That's the most stressed & desirable need in software development. I don't think it will require much effort.